daemon/archive.go

```diff
@@ -31,18 +31,19 @@ type archiver interface {
 }
 
 // helper functions to extract or archive
-func extractArchive(i interface{}, src io.Reader, dst string, opts *archive.TarOptions) error {
+func extractArchive(i interface{}, src io.Reader, dst string, opts *archive.TarOptions, root string) error {
```

Comment:
Would it perhaps make sense to add `root` to `archive.TarOptions` instead of as a new argument?

Reply:
I don't think so, those are specific to tar format.

Reply:
Good point; I was a bit in doubt; had a quick look at other options, and wasn't sure if all the existing ones were really Tar-specific, or just a mechanism to pass options for our specific purpose